test/test-lex.cpp

```diff
@@ -136,12 +136,119 @@ TEST(test_lex, lex_binary_numbers) {
   check_single_token(u8"0B010101010101010", token_type::number);
 }
 
+// fails
+TEST(test_lex, DISABLED_fail_lex_binary_number) {
+  {
+    error_collector v;
+    padded_string input(u8"0b1.1");
+    lexer l(&input, &v);
+    EXPECT_EQ(l.peek().type, token_type::number);
+    l.skip();
+    EXPECT_EQ(l.peek().type, token_type::end_of_file);
+
+    // q(🎅🏾) have another error kind for
+    // `unexpected_character_in_binary_number?
+    EXPECT_THAT(v.errors, ElementsAre(ERROR_TYPE_FIELD(
+                              error_unexpected_characters_in_number, characters,
+                              offsets_matcher(&input, 3, 4))));
+  }
+}
+
+TEST(test_lex, lex_octal_numbers_strict) {
+  check_single_token(u8"000", token_type::number);
+  check_single_token(u8"001", token_type::number);
+  check_single_token(u8"00010101010101010", token_type::number);
+  check_single_token(u8"00010101010101010", token_type::number);
+  check_single_token(u8"051", token_type::number);
+  check_single_token(u8"0o51", token_type::number);
+  check_single_token(u8"0o123n", token_type::number);
+}
+
+TEST(test_lex, lex_octal_numbers_lax) {
+  check_single_token(u8"058", token_type::number);
+  check_single_token(u8"058.9", token_type::number);
+}
+
+// TODO(👮🏾‍♀️)doesn't work
+TEST(test_lex, fail_lex_octal_numbers_lax) {
```

Comment:
What does "lax" mean here?

Reply:
um...probably nothing?  When I have `TEST(test_lex, lex_octal_numbers_lax)`, I was thinking of things that would pass in sloppy mode but not strict mode.  I don't think that logic makes sense here.  Since they'll fail either way